ElementX/Sources/Screens/RoomScreen/View/TimelineItemStylerView.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  TimelineItemStyleView.swift
+//  ElementX
+//
+//  Created by Ismail on 21.06.2022.
+//  Copyright © 2022 Element. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+import Combine
+
+import Introspect
+
+struct TimelineItemStylerView<Header: View, Content: View>: View {
+
+    let timelineItem: EventBasedTimelineItemProtocol
+    @ViewBuilder let header: () -> Header
+    @ViewBuilder let content: () -> Content
+
+    var body: some View {
+        VStack(alignment: timelineItem.isOutgoing ? .trailing : .leading, spacing: -5) {
+            header()
+                .zIndex(1)
+            if timelineItem.isOutgoing {
+                HStack {
+                    Spacer()
+                    bubble
+                }
+                .padding(.trailing, 16)
+                .padding(.leading, 51)
+            } else {
+                bubble
+                    .padding(.leading, 16)
+                    .padding(.trailing, 51)
+            }
+        }
+    }
+
+    @ViewBuilder
+    var bubble: some View {
+        if shouldAvoidBubbling {
+            ZStack(alignment: .bottomTrailing) {
+                content()
+                    .clipped()
+                    .cornerRadius(8)
+                Text(timelineItem.timestamp)
+                    .foregroundColor(.global.white)
+                    .font(.element.caption2)
+                    .alignmentGuide(.trailing) { dimensions in
+                        dimensions[.trailing] + 8
+                    }
+                    .alignmentGuide(.bottom) { dimensions in
+                        dimensions[.bottom] + 8
+                    }
+                    .padding(4)
+                    .background(Color(white: 0, opacity: 0.7))
+                    .clipped()
+                    .cornerRadius(8)
+            }
+        } else {
+            VStack(alignment: .trailing, spacing: 4) {
+                content()
+                    .frame(minWidth: 64, alignment: .leading)
+
+                Text(timelineItem.timestamp)
+                    .foregroundColor(Color.element.tertiaryContent)
+                    .font(.element.caption2)
+            }
+            .padding(EdgeInsets(top: 8, leading: 8, bottom: 4, trailing: 8))
+            .clipped()
+            .background(Color.element.system)
+            .cornerRadius(8)
```

Comment:
I think something like this is closer to the Figma 🙂
```swift
@Environment(\.colorScheme) private var colorScheme

private var bubbleColor: Color {
    let opacity = colorScheme == .light ? 0.06 : 0.15
    return timelineItem.isOutgoing ? .element.accent.opacity(opacity) : .element.system
}
```
```suggestion
            .background(bubbleColor)
            .cornerRadius(12)
```

Reply:
Seems i've missed that at all. Thanks for the suggestion.